gilda/generate_terms.py

```diff
@@ -341,6 +352,113 @@ def generate_adeft_terms():
     return terms
 
 
+def generate_doid_terms():
+    return _generate_obo_terms('doid')
+
+
+def generate_efo_terms():
+    return _generate_obo_terms('efo')
+
+
+def generate_hp_terms():
+    return _generate_obo_terms('hp')
+
+
+def _generate_obo_terms(prefix):
```

Comment:
I've written a *lot* more rules for normalizing terms and xrefs in the https://github.com/pyobo/pyobo repo. I'm still improving it but right now it loads all of the ontologies in OBO Foundry plus some ones I converted myself and extracts identifiers, labels, synonyms, and xrefs in a pretty general way. Throughout the process I've also started [curating a database](https://github.com/pyobo/pyobo/blob/master/src/pyobo/registries/metaregistry.json) of prefixes, synonyms for prefixes, etc. that would supplement the Identifiers.org database.

I don't think you'll want to have gilda depend on that code, but maybe you can get some ideas from it. Would be happy to explain further